### PR TITLE
Add intro to scripting

### PR DESCRIPTION
**Pull request by kaidamasaki**
## Description
A few sentences describing the overall purpose of these textbook changes.

## Questions for the Author
- [ ] Have you updated the weekly learning goals in the pedagogy resource?
- [x] Have you added links to the source (lucidchart, draw.io) for any diagram?
- [ ] Have you updated the READMEs with links to new or moved resources?

## Todos
Are any of these items still in progress?
- [x] Section X still needs some more detailed examples
- [x] Still need to add "Additional Resources"

## Feedback Requested
Please utilize the built-in GitHub Reviewers functionality to assign necessary reviewers. Use this section to further specify portions of the PR that you would like additional or focused feedback on.

### CODE REVIEW

**Comment by kaidamasaki** [comment]
So, as I started to write this out it turned out to be _rather_ involved.  I'm not sure it actually makes sense to include this since the various steps are pretty involved.

File: 00-programming-fundamentals/intro-to-scripts.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+By now you should be familiar with a variety of different commands from the terminal.  Things like `ls` and `touch`.  However, it turns out macOS doesn't come with _every_ command you could ever want.  Things like the `code` and `brew` commands are once you added yourself.
+
+A "script" is a command written in a "Scripting Language," luckily you already know one, Ruby!  Scripting languages are special in that they are designed to be easy to read _and_ run.  If you were to open the `ls` command all you are would see is a bunch of gibberish.
+
+## How to run a script
+
+Start by downloading this [demo script](./demo-script.rb)!
+
+You should already be familiar with the simplest way to run a Ruby script.  You can `ruby ./demo-script.rb`.  If a script is properly formatted with a magic comment at the top you can run it with `./demo-script-rb` leaving off the `ruby` from the front.
```
Comment (droberts-ada) [inline 00-programming-fundamentals/intro-to-scripts.md]: Don't need the trailing `-rb`

File: 00-programming-fundamentals/intro-to-scripts.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# Introduction to Scripts
+
+## What is a script?
+
```
Comment (droberts-ada) [inline 00-programming-fundamentals/intro-to-scripts.md]: You should have learning goals at the top of this lesson. Something like:

```md
## Learning Goals

By the end of this lesson, students should be able to...
- Define something
- Use something to something
```

File: 00-programming-fundamentals/intro-to-scripts.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+To give your script permission to run you need to give it execute permission.  You can do this with:
+
+```
+$ chmod +x my-new-script.rb
+
+# Now if you try to run my-new-script.rb it works!
+$ ./my-new-script.rb
+Success!
+```
```
Comment (droberts-ada) [inline 00-programming-fundamentals/intro-to-scripts.md]: You should add `Summary` and `Additional Resources` sections here. See https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/01-ruby-fundamentals/keword-arguments.md for an example of our usual lesson structure.

File: 00-programming-fundamentals/intro-to-scripts.md (hunk 1 above)
Comment (kaidamasaki) [reply 00-programming-fundamentals/intro-to-scripts.md]: That was a brain fart, it was suppose to say `.rb`.  I just wanted to be clear that it was a Ruby file.  Do you think when testing that we should have the demo script just be `demo-script` without the extension?

Comment (droberts-ada) [reply 00-programming-fundamentals/intro-to-scripts.md]: Ah, I see. I would stick with `demo-script.rb`.

**Comment by kaidamasaki** [comment]
[Add Intro to Scripting to Textbook](https://trello.com/c/iWyyhLeS/1298-add-intro-to-scripting-to-textbook)

File: 00-programming-fundamentals/intro-to-scripts.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+## Command Line Arguments and `ARGV`
+
+Having a script that always runs the same way is useful, but sometimes you want some information from the user.  We've done this in the past by using `gets` from inside of Ruby however there's another approach that's often simpler.
+
+You've used command line arguments before in the terminal.  When you run `cd Desktop` you are calling the `cd` command with the argument `Desktop`.
```
Comment (droberts-ada) [inline 00-programming-fundamentals/intro-to-scripts.md]: Might be good to make the connection to the arguments to a method. The parallel between methods and command line scripts is a really strong one, and it's always good to tie back into prior learning.

**Pull request merged**